client/components/checkout-form.jsx

```diff
@@ -0,0 +1,255 @@
+      {({ stripe, elements }) => (
+        <Checkout stripe={stripe} elements={elements} />
+      )}
+    </ElementsConsumer>
+  );
+}
+
+class Checkout extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      checkout: 'contactInfo',
+      email: '',
+      firstName: '',
+      lastName: '',
+      address: '',
+      address2: '',
+      city: '',
+      state: '',
+      zip: '',
+      costs: {}
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.nextButton = this.nextButton.bind(this);
+    this.payment = this.payment.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const token = window.localStorage.getItem('token');
+    fetch('api/cost', {
+      method: 'GET',
+      headers: {
+        'X-Access-Token': token
+      }
+    })
+      .then(res => res.json())
+      .then(costs => this.setState({ costs }))
+      .catch(err => console.error(err));
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  payment() {
+    if (!this.state.email || !this.state.firstName || !this.state.lastName || !this.state.address ||
+        !this.state.city || !this.state.state || !this.state.zip) {
+      alert('Please fill out the missing information.');
+    } else {
+      this.setState({ checkout: 'payment' });
+    }
+  }
+
+  nextButton() {
+    if (this.state.checkout === 'contactInfo') {
+      return (
+        <div className="col-md-4 contact-info-summary">
+          <OrderSummary subtotal={this.state.costs.subtotal} taxes={`$${this.state.costs.taxes}`} total={this.state.costs.total}/>
+          <div className="d-flex justify-content-center">
+            <Button type="button" className="mx-3 mb-3 w-100 ctn-to-payment-btn border-0" onClick={this.payment}>Continue to Payment</Button>
+          </div>
+        </div>
+      );
+    }
+    return null;
+  }
+
+  handleSubmit = async event => {
+    event.preventDefault();
+    const token = window.localStorage.getItem('token');
+    const address2 = this.state.address2 === '' ? null : this.state.address2;
+    const body = {
+      email: this.state.email,
+      firstName: this.state.firstName,
+      lastName: this.state.lastName,
+      address: this.state.address,
+      address2,
+      city: this.state.city,
+      state: this.state.state,
+      zip: this.state.zip,
+      total: 45678
+    };
+    fetch('/api/checkout', {
+      method: 'POST',
+      headers: {
+        'X-Access-Token': token,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(body)
+    })
+      .then(res => res.json())
+      .catch(err => console.error(err));
+
+    const { stripe, elements } = this.props;
+
+    if (!stripe || !elements) {
+      return;
+    }
+
+    stripe.confirmPayment({
+      elements,
+      confirmParams: {
+        return_url: 'http://localhost:3000/#home'
```

Comment:
We should avoid hard coding our host, since we need to deploy to another host later. `'/#home'` should suffice.